R/FilteredData.R

```diff
@@ -908,6 +867,51 @@ FilteredData <- R6::R6Class( # nolint
 
   ## __Private Methods ====
   private = list(
+    # Handles changes in `active_datanames`.
+    # @param active_datanames (`reactive(1)`) the observed reactive
+    # @return a `shiny` observer
+    active_datanames_observer = function(active_datanames) {
+      # rather than regenerating the UI dynamically for the dataset filtering,
+      # we instead choose to hide/show the elements
+      # the filters for this dataset are just hidden from the UI, but still applied
+      # optimization: we set `priority = 1` to execute it before the other
+      # observers (default priority 0), so that they are not computed if they are hidden anyways
+      observeEvent(active_datanames(),
+        priority = 1,
+        {
+          logger::log_trace(
+            "FilteredData$srv_filter_panel@1 active datanames: { paste(active_datanames(), collapse = \" \") }"
+          )
+
+          private$hide_inactive_datasets(active_datanames)
+          if (length(active_datanames()) == 0 || is.null(active_datanames())) {
+            # hide the filter panel UI when the active module does not use any datasets from the panel
+            shinyjs::runjs("if (filter_open) toggle_sidebar();")
+          }
```

Comment:
Make note of this reversed order of business here compared to `main`. 

Here we:
1. hide inactive dataset names from the panel
2. Hide the panel

On main we do the reverse which leads to the lack of update of the names in the filter panel because `shiny` does not see a need to update anything in the UI if it is hidden anyway. Sadly, when the UI is dynamically shown, it does not detect that these inactive dataset names should be removed from the UI. That lead to the active datanames being a copy of the last module's that had active datanames.